### ISSUE TITLE
Cairo Module not found after installing dependencies

### ISSUE THREAD

**Issue by kkin1995** [3b1b/manim#957]
Hi, I installed manim and all other dependencies including cairo, pycairo and cairo pkg-cofig (it was a requirement for pycairo). Then I ran the command:

`manim my_project.py MyScene`

as shown in the manim README file and got the following error. 

`Traceback (most recent call last):
  File "/usr/local/bin/manim", line 6, in <module>
    from manimlib import main
  File "/Library/Python/3.7/site-packages/manimlib/__init__.py", line 4, in <module>
    import manimlib.extract_scene
  File "/Library/Python/3.7/site-packages/manimlib/extract_scene.py", line 9, in <module>
    from manimlib.scene.scene import Scene
  File "/Library/Python/3.7/site-packages/manimlib/scene/scene.py", line 11, in <module>
    from manimlib.camera.camera import Camera
  File "/Library/Python/3.7/site-packages/manimlib/camera/camera.py", line 9, in <module>
    import cairo
ModuleNotFoundError: No module named 'cairo'`

Additionally, I also tried to import cairo from python interactive mode which gave the same error:

`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named cairo`

Any idea how to resolve this issue?

**Comment by stevenschn** [comment]
Did you already try to just manually pip install the following cairo version:
pycairo-1.19.1-cp37-cp37m-win32

Hence, download that file (google will give you a link), run CMD from that folder and type
`py pip install pycairo-1.19.1-cp37-cp37m-win32.whl`

Some computers may take `python` or `python3` instead of `py`.
Note: this exact version is for windows (not being AMD64) and python 3.7.

**Comment by kkin1995** [comment]
Hi, I did install pycairo version 1.19.1 and the requirement cairo pkg-config using the commands:

`pip3 install pycairo`
`brew install cairo pkg-config`

Also, apologies, I should have mentioned in my original post that I am using a mac.

**Comment by stevenschn** [comment]
I would uninstall it and try the 1.18 version. At first I tried all the newest versions as well, but I guess the dependencies do not work out well. Hence, try the 1.18 version.

**Comment by kkin1995** [comment]
I reinstalled pycairo version 1.18 but unfortunately it's giving the same issue. Also, I have cairo version 1.16.0 and pkg_config version 0.29.2. And, i am not sure if this is connected to this issue, but when i run the following command:

`python3 -m manim example_scenes.py SquareToCircle -pl`

Python gives me an error which says that the manim package is not found.

**Comment by stevenschn** [comment]
That's unfortunate. I can only tell you my experience from windows.

Can you confirm manimlib is installed? Thus, if you type `python3 pip install manimlib`  (or `pip3 install manimlib` or `pip install manimlib`) does it return it is already installed? And does the same apply when the requirements, hence do you get a confirmation after running the prompt from your manim folder and typing `pip3 install -r requirements.txt`?

**Comment by kkin1995** [comment]
Hi, Yes, Manim is installed. It was the first package that I installed using `pip3 install manimlib`. And when I run `pip3 install -r requirements.txt`, I get the following message which says that all requirements have been satisfied:

`Ignoring pycairo: markers 'sys_platform == "linux"' don't match your environment
Ignoring pycairo: markers 'sys_platform == "win32"' don't match your environment
Ignoring pyreadline: markers 'sys_platform == "win32"' don't match your environment
Requirement already satisfied: argparse==1.4.0 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 1)) (1.4.0)
Requirement already satisfied: colour==0.1.5 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 2)) (0.1.5)
Requirement already satisfied: numpy==1.16.4 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 3)) (1.16.4)
Requirement already satisfied: Pillow==5.2.0 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 4)) (5.2.0)
Requirement already satisfied: progressbar==2.5 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 5)) (2.5)
Requirement already satisfied: scipy==1.3.0 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 6)) (1.3.0)
Requirement already satisfied: tqdm==4.24.0 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 7)) (4.24.0)
Requirement already satisfied: opencv-python==3.4.2.17 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 8)) (3.4.2.17)
Requirement already satisfied: pydub==0.23.0 in /usr/local/lib/python3.7/site-packages (from -r requirements.txt (line 11)) (0.23.0)`

Additionally, to confirm, I also tried running the command `pip3 install manimlib` which gave a similar message:

`Requirement already satisfied: manimlib in /usr/local/lib/python3.7/site-packages (0.1.10)
Collecting numpy==1.15.0
  Using cached numpy-1.15.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (24.5 MB)
Requirement already satisfied: colour==0.1.5 in /usr/local/lib/python3.7/site-packages (from manimlib) (0.1.5)
Requirement already satisfied: opencv-python==3.4.2.17 in /usr/local/lib/python3.7/site-packages (from manimlib) (3.4.2.17)
Requirement already satisfied: progressbar==2.5 in /usr/local/lib/python3.7/site-packages (from manimlib) (2.5)
Collecting scipy==1.1.0
  Using cached scipy-1.1.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (16.7 MB)
Requirement already satisfied: pydub==0.23.0 in /usr/local/lib/python3.7/site-packages (from manimlib) (0.23.0)
Requirement already satisfied: Pillow==5.2.0 in /usr/local/lib/python3.7/site-packages (from manimlib) (5.2.0)
Requirement already satisfied: argparse==1.4.0 in /usr/local/lib/python3.7/site-packages (from manimlib) (1.4.0)
Requirement already satisfied: tqdm==4.24.0 in /usr/local/lib/python3.7/site-packages (from manimlib) (4.24.0)
Installing collected packages: numpy, scipy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.16.4
    Uninstalling numpy-1.16.4:
      Successfully uninstalled numpy-1.16.4
  Attempting uninstall: scipy
    Found existing installation: scipy 1.3.0
    Uninstalling scipy-1.3.0:
      Successfully uninstalled scipy-1.3.0
Successfully installed numpy-1.15.0 scipy-1.1.0`

**Comment by stevenschn** [comment]
That indeed all looks good, but since your error is about cairo, that is where your problem is. You could try this tutorial, a friend of mine actually did the installation today and got it working:

https://www.youtube.com/watch?v=uZj_GQc6pN4&list=PL2B6OzTsMUrwo4hA3BBfS7ZR34K361Z8F&index=4&t=0s

**Comment by kkin1995** [comment]
Hi, Thanks for all your help ! I figured out what was wrong and I discovered that it was my mistake and when I was executing the program, it was searching for the package (cairo) in /usr/local/bin but I had installed the package in the manim folder itself as part of a virtual environment.

In short, when I try to execute the python file using the command `manim` it does not work whereas if I navigate to the manim folder and then execute it works.

I shall close this issue with this comment.
